lib/rucio/daemons/oauthmanager/oauthmanager.py

```diff
@@ -38,140 +38,139 @@
-
-        except (DatabaseException, DatabaseError) as err:
-            if match('.*QueuePool.*', str(err.args[0])):
-                logger(logging.WARNING, traceback.format_exc())
-                record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
-            elif match('.*ORA-03135.*', str(err.args[0])):
-                logger(logging.WARNING, traceback.format_exc())
-                record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
-            else:
-                logger(logging.CRITICAL, traceback.format_exc())
-                record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
-        except Exception as err:
-            logger(logging.CRITICAL, traceback.format_exc())
-            record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
+    graceful_stop.wait(1)
+
+    start = time.time()
+
+    ndeleted = 0
+    ndeletedreq = 0
+    nrefreshed = 0
 
-        try:
-            # waiting 1 sec as DBs does not store milisecond and tokens
-            # eligible for deletion after refresh might not get dleeted otherwise
-            GRACEFUL_STOP.wait(1)
-            # EXPIRED TOKEN DELETION
-            logger(logging.INFO, '----- START ----- DELETION OF EXPIRED TOKENS ----- ')
-            logger(logging.INFO, 'starting to delete expired tokens')
-            ndeleted += delete_expired_tokens(heartbeat['nr_threads'], heartbeat['assign_thread'] + 1, limit=max_rows)
-            logger(logging.INFO, 'deleted %i expired tokens', ndeleted)
-            logger(logging.INFO, '----- END ----- DELETION OF EXPIRED TOKENS ----- ')
-            record_counter(name='oauth_manager.tokens.deleted', delta=ndeleted)
-
-        except (DatabaseException, DatabaseError) as err:
-            if match('.*QueuePool.*', str(err.args[0])):
-                logger(logging.WARNING, traceback.format_exc())
-                record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
-            elif match('.*ORA-03135.*', str(err.args[0])):
-                logger(logging.WARNING, traceback.format_exc())
-                record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
-            else:
-                logger(logging.CRITICAL, traceback.format_exc())
-                record_counter('oauth_manager.exceptions.{exception}', labels={'exception': err.__class__.__name__})
-        except Exception as err:
+    try:
+        # ACCESS TOKEN REFRESH - better to run first (in case some of the refreshed tokens needed deletion after this step)
+
+        # make a heartbeat
+        worker_number, total_workers, logger = heartbeat_handler.live()
+
+        logger(logging.INFO, '----- START ----- ACCESS TOKEN REFRESH ----- ')
```

Comment:
This is unrelated to this PR. So you can leave it unchanged, but splitting one log message into two separate lines like that is not a good idea. It may improve the readability of logs for people who look at bare logs, but it makes things worse for everybody who uses a proper log collection pipeline and then searches for logs in elastic-search for example.

Reply:
What you say is true, this again I left untouched, I would probably assign this logging refactoring to a separate issue/PR.